content/How to change the default datepicker on the Django admin page - Jorge Michelena.md

```diff
@@ -0,0 +1,144 @@
+
+
+```python
+from datetime import date
+from django import forms
+
+
+class CustomDatePickerWidget(forms.DateInput):
+
+    DATE_INPUT_WIDGET_REQUIRED_FORMAT = '%Y-%m-%d'
+
+    def __init__(self, attrs={}, format=None):
+        attrs.update(
+            {
+                'class': 'form-control',
+                'type': 'date',
+            }
+        )
+        self.format = format or self.DATE_INPUT_WIDGET_REQUIRED_FORMAT
+        super().__init__(attrs, format=self.format)
+
+```
+
+Here we are defining what date format the input is gonna enforce (`%Y-%m-%d` meaning `year-month-day`) and in the class constructor we are doing two things:
+- Adding `class`: `form-control` and `type`: `date` to `attrs` which is important to ensure that a calendar widget will be displayed from where we can select a date (if we didn't the input displayed for this field in the form would be a plain text one)
+- Setting `%Y-%m-%d` as the class' `format` attribute if no other format is provided as an argument.
+Then we are simply calling the parent class' constructor to handle the rest.
+
+With this we already have a widget that addresses our main concern of navigating through years in a convenient way, you can subclass
+it to more specific properties to better handle your use case. For our `Customer` model since we will be handling birth dates we also want to make sure that no future dates will be used as input, so in the same file just under this class we will define:
+
+
+```python
+...
+
+class PastCustomDatePickerWidget(CustomDatePickerWidget):
+
+    def __init__(self, attrs={}, format=None):
+        attrs.update({'max': date.today()})
+        super().__init__(attrs, format=format)
+
+```
+
+As you can see in the constructor we update `attrs` specifying `max` as `date.today()`.
+Now, with our new widget defined we just need to use it where we registered the model in the admin page to override the default one.
+At `project/customers/admin.py` I already had a `CustomerAdmin` which was the class I used to register the `Customer` model on the
+admin page. In my case it was empty so I am going to add the following code:
+
+
+```python
+from django.contrib import admin
+from django.db import models
+from .models import Customer
+from project.widgets import PastCustomDatePickerWidget
+
+
+@admin.register(Customer)
+class CustomerAdmin(admin.ModelAdmin):
+    formfield_overrides = {
+        models.DateField: {'widget': PastCustomDatePickerWidget},
+    }
+
+
+...
+
+```
+
+What we are doing here is importing the datepicker and using it as the value for `widget` in the `formfield_overrides` as shown in the code.
+
+![image](images/new_datepicker.gif)
+
+And that's it! Now we can input birth date values in a much more comfortable manner instead of scrolling through months endlessly!
+
+
+### Conclusion
+
+In this blog post, you learned how to create a new custom datepicker for Django as well as how to use it as the default datepicker on the Django admin page.
```

Comment:
I would add something like "thank you for reading" something like that. Some phrase having that idea